src/tools/kata-ctl/src/check.rs

```diff
@@ -89,13 +95,49 @@ pub fn run_network_checks() -> Result<()> {
     Ok(())
 }
 
+fn get_kata_version_by_url(url: &str) -> Result<String, reqwest::Error> {
+    let content = reqwest::blocking::Client::new()
+        .get(url)
+        .header(CONTENT_TYPE, "application/json")
+        .header(USER_AGENT, "kata")
+        .send()?
+        .json::<HashMap<String, Value>>()?;
+
+    let version = content["tag_name"].as_str().unwrap();
+    Ok(version.to_string())
+}
+
+fn handler(a: reqwest::Error) {
+    if a.is_connect() {
+        println!("http connection failure: connection refused");
+    }
+
+    if a.is_timeout() {
+        println!("http connection failure: connection timeout");
+    }
+
+    if a.is_builder() {
+        println!("http connection failure: url malformed");
+    }
+
+    if a.is_decode() {
+        println!("http connection failure: unable to decode response body");
+    }
+}
```

Comment:
This should really return the error and also not print to stdout. Something like this (you'll need to add `use anyhow::Context;` too btw):

```suggestion
fn handle_reqwest_error(e: reqwest::Error) -> anyhow::Error {
    if e.is_connect() {
        return anyhow!(e).context("http connection failure: connection refused");
    }

    if e.is_timeout() {
        return anyhow!(e).context("http connection failure: connection timeout");
    }

    if e.is_builder() {
        return anyhow!(e).context("http connection failure: url malformed");
    }

    if e.is_decode() {
        return anyhow!(e).context("http connection failure: unable to decode response body");
    }

    anyhow!(e).context("unknown http connection failure: {:?}")
}
```

Reply:
@jodh-intel thank you for your feedback